### PR TITLE
Fix `numpy.random` seed when testing Parameter Shift Rule.

### PR DESCRIPTION
**Pull request by MatteoRobbiati**
Fixing #1181. 
Thanks @renatomello for raising the issue. Fixing the random generator seed should prevent such failed tests.

Checklist:
- [ ] Reviewers confirm new code works as expected.
- [x] Tests are passing.
- [x] Coverage does not decrease.
- [x] Documentation is updated.

### CODE REVIEW

**Review by renatomello — DISMISSED** [review]

**Comment by renatomello** [comment]
Does fixing the seed using `numpy` also solves the issue for GPU backends?

File: tests/test_derivative.py [hunk 1]
```diff
@@ -33,6 +33,7 @@ def circuit(nqubits=1):
 def test_standard_parameter_shift(backend, nshots, atol, scale_factor, grads):
     # initializing the circuit
     c = circuit(nqubits=1)
+    np.random.seed(42)
```
Comment (renatomello) [inline tests/test_derivative.py:36]:
```suggestion
    backend.set_seed(42)
```

Maybe this is more general

File: tests/test_derivative.py [hunk 2]
```diff
@@ -88,6 +89,7 @@ def test_standard_parameter_shift(backend, nshots, atol, scale_factor, grads):
 def test_finite_differences(backend, step_size):
     # initializing the circuit
     c = circuit(nqubits=1)
+    np.random.seed(42)
```
Comment (renatomello) [inline tests/test_derivative.py:92]:
```suggestion
    backend.set_seed(42)
```

**Comment by MatteoRobbiati** [comment]
> Does fixing the seed using `numpy` also solves the issue for GPU backends?

Oh yes, you are right! Thanks!

**Review by renatomello — APPROVED** [review]

**Review by BrunoLiegiBastonLiegi — APPROVED** [review]

**Pull request merged**